R/resource.R

```diff
@@ -0,0 +1,160 @@
+##' Specify what resources a task requires to run.
+##'
+##' # Windows cluster
+##'
+##' The windows cluster currently is all 32-core nodes, with RAM
+##' mainly 512Gb, and a few 384Gb.
+##'
+##'
+##' # Linux cluster
+##'
+##' Coming Soon.
+##'
+##' @title Task Resources
+##'
+##' @param cores The number of cores your task requires. This is 1 by
+##'   default. An error is reported if you request more cores than
+##'   is available on the largest cluster node. Setting to `Inf` will
+##'   request any entire node, however many cores it has.
+##'
+##' @param exclusive Set this to `TRUE` to ensure no other tasks will be
+##'   concurrently run on the node while it runs your task.
+##'
+##' @param runtime Set this to specify a time limit for running your job.
+##'   Acceptable formats are either an integer number of minutes, or
+##'   strings in the form `"h:m"`, or `"d:h:m"` to specify
+##'   days, hours and minutes.
+##'
+##' @param hold_until Specify your task should wait in the queue until
+##'   a certain time, or for a certain period. For the former, this can
+##'   be a POSIXt, a Date, the special strings "tonight" (7pm), "midnight",
+##'   or "weekend" (midnight Saturday morning). To delay for a period,
+##'   this can be an integer number of minutes, or a string in the form
+##'   `"h:m"` for hours and minutes.
+##'
+##' @param memory_per_node Specify your task can only run on a node
+##'   with at least the specified memory. This is an integer,
+##'   measured in gigabytes, or strings such as `"64Gb"` or `"1Tb"`.
+##'
+##' @param memory_per_process If you can provide an estimate of how
+##'   much RAM your task requires, then the cluster can ensure the
+##'   total memory required by running multiple tasks on a node
+##'   does not exceed how much memory the node has. Specify this as
+##'   an integer number of gigabytes, or characters such as `"10Gb"`
+##'
+##' @param requested_nodes If you have been in touch with us or DIDE
+##'   IT, and you need to run your task on a selection of named compute
+##'   nodes, then specify this here as a vector of strings for the node names.
+##'
+##' @param priority If the tasks you are launching are low priority, you can
+##'   allow other queuing tasks to jump over them, by setting the priority to
+##'   to `low`; otherwise, the default is `normal`. These are the only
+##'   acceptable values.
+##'
+##' @param queue Specify a particular queue to submit your tasks to.
+##'   This is in development as we decide over time what
+##'   queues we best need for DIDE's common workflows. See the Details
+##'   for more information
+##'
+##' @return A hipercow_resource list, which can be provided when you submit
```

Comment:
Given this returns a an object of class `hipercow_resource`, it really should be called `hipercow_resource`, not `task_resource`
```suggestion
##' @return A `hipercow_resource` list, which can be provided when you submit
```